doc/api/assert.md

```diff
@@ -330,6 +330,11 @@ changes:
 Asserts that the function `block` does not throw an error. See
 [`assert.throws()`][] for more details.
 
+Please note: Using `assert.doesNotThrow()` is not recommended because there is
+no benefit by catching an error and then rethrowing it. Instead, consider adding
```

Comment:
Nit: `by` -> `to`

Reply:
s/recommented/actually useful/

Given that we are not deprecating it and that there is no direct harm, I don't see a reason to recommend avoiding it.
A note that it doesn't do anything useful seems better fit, imo.

Reply:
I think we can do both. Say that it is not useful and therefore not recommended to use.